config/README.md

```diff
@@ -166,6 +167,226 @@ fmt.Println(m.World)
+    "Op": "*", "Left": "3", "Right": "5"
+  }))
+
+  require.Error(t, err)
+  assert.Contains(t, err.Error(), `unknown operation "*"`)
+}
+```
+
+For integration/E2E testing you can customize `config.Loader` to load
+configuration files from either custom folders(`Loader.SetDirs()`),
+or custom files(`Loader.SetFiles()`), or register new providers on top of
+existing providers(`Loader.RegisterProviders()`) that will override values
+of default configs.
+
+## Utilities
+
+`Config` package comes with several helpers that can make writing tests,
+create new providers or amend existing ones much easier.
+
+* `NewCachedProvider(p Provider)` returns a new provider that wraps `p`
+  and caches values in underlying map. It also registers callbacks to track
+  changes in all values it cached, so you can call `cached.Get("something")`
+  and don't worry about latencies much. It is safe for concurrent use by
+  multiple goroutines.
+
+* `MockDynamicProvider` is a mock provider that can be used to test dynamic
+  features, it implements `Provider` interface and lets you to set values
+  to trigger change callbacks.
+
+* Sometimes dynamic providers let you to register only one callback per key.
+  If you want to have multiple keys per callback you can use
+  `NewMultiCallbackProvider(p Provider)` wrapper, that will store a list of
+  all callbacks for each value and call them when a value changes.
+  Caution: it locks provider during callbacks execution, you should try to
+  make this callbacks as fast as possible.
+
+* `NopProvider` is useful for testing because it can be embedded in any type
+  if you are not interested in implementing all Provider methods.
+
+* `NewProviderGroup(name string, providers ...Provider)` groups providers in one.
+  Lookups for values are determined by the order providers passed:
+  `NewProviderGroup("global", provider1, provider2)`, first `provider1` will be
+  checked and if there is no value, it will return `provider2.Get()`.
+
+* `NewStaticProvider(data interface{})` is very a useful wrapper for testing,
+  you can pass custom maps and use them as configs instead of loading them
+  from files.
+
+## Loading Configuration
+
+Load process is controlled by `config.Loader`. If a service doesn't specify a
+config provider, manager is going to use a provider returned by
+`config.DefaultLoader.Load()`.
+
+The default loader will load static providers first:
+
+* YAML provider will look for `base.yaml` and `${environment}.yaml` files in
+  current folder and then in `./config` folder. You can override folders
+  to look for these files with `Loader.SetDirs()`.
+  To override files names use `Loader.SetFiles()`.
+
+* Command line provider will look for `--roles` argument to specify service
+  roles. You can introduce/override config values by adding new flags to
+  `pflags.CommandLine` set before building a service.
+
+You can add more static providers on top of mentioned above with
+`RegisterProviders()` function:
```

Comment:
Ah sorry.
with `RegisterProviders()` --> with the `RegisterProviders()`

Extra notes for Alex: Any time you have "X function," "X set," "X variable," etc., make sure you add "the" before it, like this: "the X function," "the X set," "the X variable." If you're just saying X alone, it's OK without "the."

I have plenty of examples if you're confused about this rule.